gen/calc.py

```diff
@@ -575,6 +575,30 @@ def fz_err(msg):
             validate_int_in_range(port, 1, 65535)
 
 
+def calculate_scarce_resources(enable_scarce_resources):
+    return ','.join(enable_scarce_resources)
+
+
+def calculate_filter_gpu_reources(enable_scarce_resources):
+    if len(enable_scarce_resources) == 0:
+        return 'false'
+    if enable_scarce_resources[0] == 'gpus':
+        return 'true'
+    return 'false'
+
+
+def validate_scarce_resources(enable_scarce_resources):
+    validate_json_list(enable_scarce_resources)
+    if len(enable_scarce_resources) == 0:
+        return
+
+    assert len(enable_scarce_resources) != 1, '{} not length 1'.format(enable_scarce_resources)
```

Comment:
This assertion will fail when the length is 1, so that error message would be confusing since it's stating the opposite.